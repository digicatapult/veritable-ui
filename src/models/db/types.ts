import { Knex } from 'knex'
import { z } from 'zod'
import { carbonEmbodimentRequestData, carbonEmbodimentResponseData } from '../drpc.js'

export const tablesList = [
  'connection',
  'connection_invite',
  'query',
  'query_rpc',
  'settings',
  'organisation_registries',
] as const

const insertConnection = z.object({
  company_name: z.string(),
  company_number: z.string(),
  status: z.union([
    z.literal('pending'),
    z.literal('unverified'),
    z.literal('verified_them'),
    z.literal('verified_us'),
    z.literal('verified_both'),
    z.literal('disconnected'),
  ]),
  agent_connection_id: z.union([z.string(), z.null()]),
  pin_attempt_count: z.number().int().gte(0).lte(255),
  pin_tries_remaining_count: z.number().int().gte(0).lte(255).nullable(),
  registry_country_code: z.string(),
})

const insertOrganisationRegistries = z.object({
  country_code: z.string(),
  registry_name: z.string(),
  registry_key: z.string(),
  url: z.string(),
  api_key: z.string(),
})
const insertConnectionInvite = z.object({
  connection_id: z.string(),
  oob_invite_id: z.string(),
  pin_hash: z.string(),
  expires_at: z.date(),
  validity: z.union([z.literal('valid'), z.literal('expired'), z.literal('too_many_attempts'), z.literal('used')]),
})

const insertQueryShared = z.object({
  connection_id: z.string(),
  parent_id: z.string().nullable().optional(),
  type: z.literal('total_carbon_embodiment'),
  status: z.enum(['resolved', 'pending_your_input', 'pending_their_input', 'errored', 'forwarded']),
  response_id: z.string().nullable(),
  role: z.enum(['requester', 'responder']),
  expires_at: z.date(),
})
const totalCarbonEmbodimentQuery = z.object({
  type: z.literal('total_carbon_embodiment'),
  details: carbonEmbodimentRequestData,
  response: carbonEmbodimentResponseData.nullable(),
})
const insertQuery = z.discriminatedUnion('type', [totalCarbonEmbodimentQuery.merge(insertQueryShared)])

const defaultFields = z.object({
  id: z.string(),
  created_at: z.date(),
  updated_at: z.date(),
})

const insertQueryRpc = z.object({
  query_id: z.string(),
  agent_rpc_id: z.string(),
  role: z.union([z.literal('client'), z.literal('server')]),
  method: z.union([z.literal('submit_query_request'), z.literal('submit_query_response')]),
  result: z.union([z.record(z.any()), z.null()]).optional(),
  error: z.union([z.record(z.any()), z.null()]).optional(),
})
const insertSettings = z.object({
  setting_key: z.string(),
  setting_value: z.string(),
})

const Zod = {
  connection: {
    insert: insertConnection,
    get: insertConnection.merge(defaultFields),
  },
  connection_invite: {
    insert: insertConnectionInvite,
    get: insertConnectionInvite.merge(defaultFields),
  },
  query: {
    insert: insertQuery,
    get: z.discriminatedUnion('type', [totalCarbonEmbodimentQuery.merge(insertQueryShared).merge(defaultFields)]),
  },
  query_rpc: {
    insert: insertQueryRpc,
    get: insertQueryRpc.merge(defaultFields),
  },
  settings: {
    insert: insertSettings,
    get: insertSettings.extend({
      id: z.string(),
      created_at: z.date(),
      updated_at: z.date(),
    }),
  },
  organisation_registries: {
    insert: insertOrganisationRegistries,
    get: insertOrganisationRegistries.extend({
      id: z.string(),
    }),
  },
}

export type InsertConnection = z.infer<typeof Zod.connection.insert>
export type ConnectionRow = z.infer<typeof Zod.connection.get>
export type QueryRow = z.infer<typeof Zod.query.get>
export type SettingsRow = z.infer<typeof Zod.settings.get>
export type OrganisationRegistriesRow = z.infer<typeof Zod.organisation_registries.get>
export type TABLES_TUPLE = typeof tablesList
export type TABLE = TABLES_TUPLE[number]
export type Models = {
  [key in TABLE]: {
    get: z.infer<(typeof Zod)[key]['get']>
    insert: z.infer<(typeof Zod)[key]['insert']>
  }
}

export type ColumnsByType<M extends TABLE, T> = {
  [K in keyof Models[M]['get']]-?: Models[M]['get'][K] extends T ? K : never
}[keyof Models[M]['get']]

type WhereComparison<M extends TABLE> = {
  [key in keyof Models[M]['get']]: [
    Extract<key, string>,
    '=' | '>' | '>=' | '<' | '<=' | '<>' | 'LIKE' | 'ILIKE',
    Extract<Models[M]['get'][key], Knex.Value>,
  ]
}
export type WhereMatch<M extends TABLE> = {
  [key in keyof Models[M]['get']]?: Models[M]['get'][key]
}

export type Where<M extends TABLE> = WhereMatch<M> | (WhereMatch<M> | WhereComparison<M>[keyof Models[M]['get']])[]
export type Order<M extends TABLE> = [keyof Models[M]['get'], 'asc' | 'desc'][]
export type Update<M extends TABLE> = Partial<Models[M]['get']>

export type IDatabase = {
  [key in TABLE]: () => Knex.QueryBuilder
}

export default Zod
