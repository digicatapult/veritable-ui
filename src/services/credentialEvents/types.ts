import { Credential, CredentialFormatData } from '../../models/veritableCloudagent/internal.js'

export interface CredentialEventHandlerBase {
  handleProposalReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleOfferReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleRequestReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleCredentialReceived(credential: Credential, formatData: CredentialFormatData): Promise<void>
  handleDone(credential: Credential, formatData: CredentialFormatData): Promise<void>
}

export interface CredentialEventHandler<EventName extends string, EventVersion extends string>
  extends CredentialEventHandlerBase {
  get schemaName(): EventName
  get schemaVersion(): EventVersion
}
