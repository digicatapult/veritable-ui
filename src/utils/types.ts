export type DiscriminateUnion<T, K extends keyof T, V extends T[K]> = T extends Record<K, V> ? T : never

export type MapDiscriminatedUnion<T extends Record<K, string>, K extends keyof T> = {
  [V in T[K]]: DiscriminateUnion<T, K, V>
}
