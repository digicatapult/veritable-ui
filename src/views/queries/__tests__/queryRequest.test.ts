import { expect } from 'chai'
import { describe, test } from 'mocha'
import { ConnectionRow } from '../../../models/db/types.js'
import QueryRequestTemplates from '../queryRequest.js'

describe('QueryRequestTemplates', () => {
  describe('newQueryRequestPage', () => {
    test('companySelect with companies', async () => {
      const templates = new QueryRequestTemplates()
      const rendered = await templates.newQueryRequestPage({
        formStage: 'companySelect',
        connections: [
          {
            id: '1',
            company_name: 'company1',
            company_number: '111111111',
          },
          {
            id: '2',
            company_name: 'company2',
            company_number: '222222222',
          },
        ] as ConnectionRow[],
        search: 'search-text',
        type: 'total_carbon_embodiment',
      })
      expect(rendered).to.matchSnapshot()
    })

    test('companySelect with no companies', async () => {
      const templates = new QueryRequestTemplates()
      const rendered = await templates.newQueryRequestPage({
        formStage: 'companySelect',
        connections: [],
        search: 'search-text',
        type: 'total_carbon_embodiment',
      })
      expect(rendered).to.matchSnapshot()
    })

    test('carbon embodiment form without product and quantity inputs', async () => {
      const templates = new QueryRequestTemplates()
      const rendered = await templates.newQueryRequestPage({
        formStage: 'carbonEmbodiment',
        connectionId: 'connection-id',
        type: 'total_carbon_embodiment',
      })
      expect(rendered).to.matchSnapshot()
    })

    test('carbon embodiment form with product and quantity inputs', async () => {
      const templates = new QueryRequestTemplates()
      const rendered = await templates.newQueryRequestPage({
        formStage: 'carbonEmbodiment',
        connectionId: 'connection-id',
        productId: 'product-id',
        quantity: 123,
        type: 'total_carbon_embodiment',
      })
      expect(rendered).to.matchSnapshot()
    })

    test('bav form', async () => {
      const templates = new QueryRequestTemplates()
      const rendered = await templates.newQueryRequestPage({
        formStage: 'bav',
        connection: { id: 'connection-id', company_name: 'company-name' } as ConnectionRow,
        type: 'beneficiary_account_validation',
      })
      expect(rendered).to.matchSnapshot()
    })

    test('success', async () => {
      const templates = new QueryRequestTemplates()
      const rendered = await templates.newQueryRequestPage({
        formStage: 'success',
        company: {
          companyName: 'company1',
        },
        type: 'total_carbon_embodiment',
      })
      expect(rendered).to.matchSnapshot()
    })

    test('error', async () => {
      const templates = new QueryRequestTemplates()
      const rendered = await templates.newQueryRequestPage({
        formStage: 'error',
        company: {
          companyName: 'company1',
        },
        type: 'total_carbon_embodiment',
      })
      expect(rendered).to.matchSnapshot()
    })
  })
})
