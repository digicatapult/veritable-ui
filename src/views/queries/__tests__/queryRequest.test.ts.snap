// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryRequestTemplates newQueryRequestPage bav form 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Beneficiary Account Validation Query</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Beneficiary Account Validation Query</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new?type=beneficiary_account_validation\\">New</a><a title=\\"Beneficiary Account Validation\\" href=\\"/queries/new?type=beneficiary_account_validation\\">Beneficiary Account Validation</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div class=\\"container-query-form\\"><div class=\\"query-form-left\\"><h1>Beneficiary Account Validation</h1><p class=\\"query-form-text\\">Creates a query to verify a company's financial details.</p></div><div class=\\"query-form-right\\"><p>Request financial details from company-name</p><form id=\\"bav\\" hx-post=\\"/queries/bav\\" hx-select=\\"main > *\\" hx-target=\\"main\\" hx-swap=\\"innerHTML\\"><input type=\\"hidden\\" name=\\"connectionId\\" value=\\"connection-id\\"/><div class=\\"input-container\\"><label for=\\"datetime-input\\" class=\\"input-label\\">Request Deadline</label><input id=\\"datetime-input\\" class=\\"input-with-label\\" type=\\"datetime-local\\" name=\\"expiresAt\\" required min=\\"2025-08-06T12:00\\"/></div><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" text=\\"Submit Query\\" style=\\"filled\\"><span>Submit Query</span></button></form></div></div></div></div></main></body></html>"`;

exports[`QueryRequestTemplates newQueryRequestPage carbon embodiment form with product and quantity inputs 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Total Carbon Embodiment Query</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Select Company To Send Your Query To</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new\\">New</a><a title=\\"Total Carbon Embodiment\\" href=\\"/queries/new/carbon-embodiment\\">Total Carbon Embodiment</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div><div class=\\"container-query-form\\"><div class=\\"query-form-left\\"><h1>Total Carbon Embodiment</h1><p class=\\"query-form-text\\">Creates a query for obtaining the total carbon embodiment for a given product/component.</p></div><div class=\\"query-form-right\\"><p>Choose the product that you want to apply the query “What is the total carbon embodiment for the product/component below?” to.</p><form id=\\"carbon-embodiment\\" hx-post=\\"/queries/new/carbon-embodiment\\" hx-select=\\"main > *\\" hx-target=\\"main\\" hx-swap=\\"innerHTML\\"><input type=\\"hidden\\" name=\\"connectionId\\" value=\\"connection-id\\"/><div class=\\"input-container\\"><label for=\\"productId-input\\" class=\\"input-label\\">Product ID</label><input id=\\"productId-input\\" name=\\"productId\\" placeholder=\\"BX20001\\" class=\\"input-with-label\\" type=\\"text\\" required value=\\"product-id\\"/><p class=\\"additional-input-label\\">Product ID</p></div><div class=\\"input-container\\"><label for=\\"productQuantity-input\\" class=\\"input-label\\">Quantity</label><input id=\\"productQuantity-input\\" name=\\"quantity\\" type=\\"text\\" placeholder=\\"123\\" pattern=\\"^\\\\d+$\\" required value=\\"123\\" class=\\"input-with-label\\"/><p class=\\"additional-input-label\\">Quantity of product</p></div><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" text=\\"Submit Query\\" style=\\"filled\\"><span>Submit Query</span></button></form></div></div></div></div></div></main></body></html>"`;

exports[`QueryRequestTemplates newQueryRequestPage carbon embodiment form without product and quantity inputs 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Total Carbon Embodiment Query</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Total Carbon Embodiment Query</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new?type=total_carbon_embodiment\\">New</a><a title=\\"Total Carbon Embodiment\\" href=\\"/queries/new?type=total_carbon_embodiment\\">Total Carbon Embodiment</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div><div class=\\"container-query-form\\"><div class=\\"query-form-left\\"><h1>Total Carbon Embodiment</h1><p class=\\"query-form-text\\">Creates a query for obtaining the total carbon embodiment for a given product/component.</p></div><div class=\\"query-form-right\\"><p>Choose the product that you want to apply the query “What is the total carbon embodiment for the product/component below?” to.</p><form id=\\"carbon-embodiment\\" hx-post=\\"/queries/carbon-embodiment\\" hx-select=\\"main > *\\" hx-target=\\"main\\" hx-swap=\\"innerHTML\\"><input type=\\"hidden\\" name=\\"connectionId\\" value=\\"connection-id\\"/><div class=\\"input-container\\"><label for=\\"productId-input\\" class=\\"input-label\\">Product ID</label><input id=\\"productId-input\\" name=\\"productId\\" placeholder=\\"BX20001\\" class=\\"input-with-label\\" type=\\"text\\" required/><p class=\\"additional-input-label\\">Product ID</p></div><div class=\\"input-container\\"><label for=\\"productQuantity-input\\" class=\\"input-label\\">Quantity</label><input id=\\"productQuantity-input\\" name=\\"quantity\\" type=\\"text\\" placeholder=\\"123\\" pattern=\\"^\\\\d+$\\" required class=\\"input-with-label\\"/><p class=\\"additional-input-label\\">Quantity of product</p></div><div class=\\"input-container\\"><label for=\\"datetime-input\\" class=\\"input-label\\">Request Deadline</label><input id=\\"datetime-input\\" class=\\"input-with-label\\" type=\\"datetime-local\\" name=\\"expiresAt\\" required min=\\"2025-08-06T12:00\\"/></div><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" text=\\"Submit Query\\" style=\\"filled\\"><span>Submit Query</span></button></form></div></div></div></div></div></main></body></html>"`;

exports[`QueryRequestTemplates newQueryRequestPage companySelect with companies 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Total Carbon Embodiment Query</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Total Carbon Embodiment Query</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new?type=total_carbon_embodiment\\">New</a><a title=\\"Total Carbon Embodiment\\" href=\\"/queries/new?type=total_carbon_embodiment\\">Total Carbon Embodiment</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div><div class=\\"main-list-page\\"><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Select a Company to send Query to</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"search-text\\" placeholder=\\"Search\\" hx-get=\\"/queries/new?type=total_carbon_embodiment\\" hx-trigger=\\"input changed delay:50ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><form id=\\"company-form\\" hx-get=\\"/queries/new?type=total_carbon_embodiment\\" hx-select=\\"main > *\\" hx-target=\\"main\\" hx-swap=\\"innerHTML\\" hx-push-url=\\"true\\"><table class=\\"list-page\\"><thead><th width=\\"4%\\"><span>Select</span><a class=\\"list-table icon disabled\\"></a></th><th width=\\"96%\\"><span>Company Name</span><a class=\\"list-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td><input type=\\"checkbox\\" class=\\"company-checkbox\\" name=\\"connectionId\\" value=\\"1\\" disabled aria-label=\\"select connection id\\"/></td><td>company1</td></tr><tr><td><input type=\\"checkbox\\" class=\\"company-checkbox\\" name=\\"connectionId\\" value=\\"2\\" disabled aria-label=\\"select connection id\\"/></td><td>company2</td></tr></tbody></table><button type=\\"submit\\" class=\\"button\\" id=\\"company-selected-next-button\\">Next</button></form></div></div></div></div></div></main></body></html>"`;

exports[`QueryRequestTemplates newQueryRequestPage companySelect with no companies 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Total Carbon Embodiment Query</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Total Carbon Embodiment Query</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new?type=total_carbon_embodiment\\">New</a><a title=\\"Total Carbon Embodiment\\" href=\\"/queries/new?type=total_carbon_embodiment\\">Total Carbon Embodiment</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div><div class=\\"main-list-page\\"><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Select a Company to send Query to</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"search-text\\" placeholder=\\"Search\\" hx-get=\\"/queries/new?type=total_carbon_embodiment\\" hx-trigger=\\"input changed delay:50ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><form id=\\"company-form\\" hx-get=\\"/queries/new?type=total_carbon_embodiment\\" hx-select=\\"main > *\\" hx-target=\\"main\\" hx-swap=\\"innerHTML\\" hx-push-url=\\"true\\"><table class=\\"list-page\\"><thead><th width=\\"4%\\"><span>Select</span><a class=\\"list-table icon disabled\\"></a></th><th width=\\"96%\\"><span>Company Name</span><a class=\\"list-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>No connections for that search. Please establish a connection and try again.</td></tr></tbody></table><button type=\\"submit\\" class=\\"button\\" id=\\"company-selected-next-button\\">Next</button></form></div></div></div></div></div></main></body></html>"`;

exports[`QueryRequestTemplates newQueryRequestPage error 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Total Carbon Embodiment Query</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Total Carbon Embodiment Query</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new?type=total_carbon_embodiment\\">New</a><a title=\\"Total Carbon Embodiment\\" href=\\"/queries/new?type=total_carbon_embodiment\\">Total Carbon Embodiment</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div id=\\"new-query-confirmation-text\\"><p>An unknown error occurred whilst submitting your query to: company1.</p><p>Please try again later.</p><a class=\\"button \\" href=\\"/queries\\" data-variant=\\"filled\\"><span>Back to Queries</span></a></div></div></div></main></body></html>"`;

exports[`QueryRequestTemplates newQueryRequestPage success 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><title>Veritable - New Total Carbon Embodiment Query</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" data-active style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Total Carbon Embodiment Query</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a><a title=\\"New\\" href=\\"/queries/new?type=total_carbon_embodiment\\">New</a><a title=\\"Total Carbon Embodiment\\" href=\\"/queries/new?type=total_carbon_embodiment\\">Total Carbon Embodiment</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"></div><div class=\\"card-body\\"><div id=\\"new-query-confirmation-text\\"><h1>Your Query has been sent!</h1><p>Your query has been successfully shared with the following supplier:</p><i><p>company1</p></i><p>Once all responses are received, the information will be automatically gathered and shared with you. No further action is needed on your part. You can trust that the process is secure, transparent, and streamlined for your convenience.</p><p>You can check the status of your query in the Queries section of your dashboard.</p><a class=\\"button \\" href=\\"/queries\\" data-variant=\\"filled\\"><span>Back to Queries</span></a></div></div></div></main></body></html>"`;
